src/plugins/graph/models/data-configuration-model.ts

```diff
@@ -85,6 +86,18 @@ export const DataConfigurationModel = types
     get yAttributeIDs() {
       return this.yAttributeDescriptions.map((d: IAttributeDescriptionSnapshot) => d.attributeID);
     },
+    /**
+     * Return true if the given attribute ID is present in any role.
+     */
+    includesAttributeID(id: string) {
+      for (const desc of self._attributeDescriptions.values()) {
+        if (desc.attributeID===id) return true;
+      }
+      for (const desc of self._yAttributeDescriptions) {
+        if (desc.attributeID===id) return true;
+      }
+      return false;
+    },
```

Comment:
See `rolesForAttribute` and `removeAttributeFromRole` in https://github.com/concord-consortium/codap/pull/971.